src/proc.cpp

```diff
@@ -114,11 +114,10 @@ void job_t::promote() {
 }
 
 void proc_destroy() {
-    job_list_t &jobs = parser_t::principal_parser().job_list();
-    for (auto job = jobs.begin(); job != jobs.end();) {
-        debug(2, L"freeing leaked job %ls", (*job)->command_wcstr());
-        job = jobs.erase(job);
+    for (const auto job : jobs()) {
```

Comment:
Should prefer `const auto &job` here to avoid the needless inc/dec of the shared pointer.